internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go

```diff
@@ -854,6 +861,46 @@ func (c *ClusterManager) buildObservabilityCloudwatchLoggingCredentialsSecret()
 	}
 }
 
+func (c *ClusterManager) buildObservabilityEnterpriseCloudwatchLoggingCredentialsSecret(orgId string, clusterId string) *k8sCoreV1.Secret {
+	idx, enterpriseCredential := arrays.FindFirst(c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.EnterpriseCredentials, func(enterpriseCredential observatorium.ObservabilityEnterpriseCloudwatchLoggingConfigCredentials) bool {
```

Comment:
Can we refactor this logic to be a method of the `ObservabilityCloudWatchLoggingConfig` type?

Reply:
:+1: